checker/src/org/checkerframework/checker/lock/LockAnnotatedTypeFactory.java

```diff
@@ -96,28 +107,310 @@ public LockTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, LockS
-        return lockAnnos;
+    // The side effect annotations processed by the Lock Checker.
+    enum SideEffectAnnotation {
+        MAYRELEASELOCKS("@MayReleaseLocks", MayReleaseLocks.class),
+        RELEASESNOLOCKS("@ReleasesNoLocks", ReleasesNoLocks.class),
+        LOCKINGFREE("@LockingFree",LockingFree.class),
+        SIDEEFFECTFREE("@SideEffectFree",SideEffectFree.class),
+        PURE("@Pure",Pure.class);
+        final String annotation;
+        final  Class<? extends Annotation> annotationClass;
+
+        SideEffectAnnotation(String annotation, Class<? extends Annotation> annotationClass) {
+            this.annotation = annotation;
+            this.annotationClass=annotationClass;
+        }
+
+        public String getNameOfSideEffectAnnotation() {
+            return annotation;
+        }
+
+        public Class<? extends Annotation> getAnnotationClass() {
+            return annotationClass;
+        }
+
+        /**
+         * Returns true if the receiver side effect annotation is weaker
+         * than side effect annotation 'other'.
+         */
+        boolean isWeakerThan(SideEffectAnnotation other) {
+            boolean weaker = false;
+
+            switch (other) {
+                case MAYRELEASELOCKS:
+                    break;
+                case RELEASESNOLOCKS:
+                    if (this == SideEffectAnnotation.MAYRELEASELOCKS) {
+                        weaker = true;
+                    }
+                    break;
+                case LOCKINGFREE:
+                    switch (this) {
+                        case MAYRELEASELOCKS:
+                        case RELEASESNOLOCKS:
+                            weaker = true;
+                            break;
+                        default:
+                    }
+                    break;
+                case SIDEEFFECTFREE:
+                    switch (this) {
+                        case MAYRELEASELOCKS:
+                        case RELEASESNOLOCKS:
+                        case LOCKINGFREE:
+                            weaker = true;
+                            break;
+                        default:
+                    }
+                    break;
+                case PURE:
+                    switch (this) {
+                        case MAYRELEASELOCKS:
+                        case RELEASESNOLOCKS:
+                        case LOCKINGFREE:
+                        case SIDEEFFECTFREE:
+                            weaker = true;
+                            break;
+                        default:
+                    }
+                    break;
+            }
+
+            return weaker;
+        }
+
+        static SideEffectAnnotation weakest = null;
+        public static SideEffectAnnotation weakest() {
+            if(weakest == null) {
+                for (SideEffectAnnotation sea : SideEffectAnnotation.values()) {
+                    if (weakest == null) weakest = sea;
```

Comment:
Fixed. I typically add braces to all blocks. Some of this code comes from Suzanne - I think she uses a slightly different coding style.